pkg/build/nodeimage/internal/kube/builder_docker.go

```diff
@@ -101,29 +100,33 @@ func (b *dockerBuilder) Build() (Bits, error) {
 		},
 		os.Environ()...,
 	)
-	// build binaries
-	what := []string{
-		// binaries we use directly
-		"cmd/kubeadm",
-		"cmd/kubectl",
-		"cmd/kubelet",
-	}
-	cmd := exec.Command(
-		"build/run.sh",
-		"make", "all", "WHAT="+strings.Join(what, " "),
-	).SetEnv(env...)
-	exec.InheritOutput(cmd)
-	if err := cmd.Run(); err != nil {
-		return nil, errors.Wrap(err, "failed to build binaries")
-	}
 
 	// build images
-	cmd = exec.Command("make", "quick-release-images").SetEnv(env...)
-	exec.InheritOutput(cmd)
-	if err := cmd.Run(); err != nil {
+	// NOTE: we do this first because it will call `make generated_files`
+	// and we're going to skip calling that when building the binaries
+	if err := exec.InheritOutput(
+		exec.Command("make", "quick-release-images").SetEnv(env...),
+	).Run(); err != nil {
 		return nil, errors.Wrap(err, "failed to build images")
 	}
 
+	// build binaries
+	if err := exec.InheritOutput(
+		exec.Command(
+			// build dockerized
+			"build/run.sh",
+			// bypass make targets to skip running codegen again, call build
+			// rule implementation directly
+			"hack/make-rules/build.sh",
```

Comment:
the "all" (default) make target is just:
```
all: generated_files
	hack/make-rules/build.sh $(WHAT)
```

and has been for over four years. https://github.com/kubernetes/kubernetes/blame/184c6d348c80c1f529b742c8959677817689a182/build/root/Makefile#L91

Reply:
at least five years: https://github.com/kubernetes/kubernetes/blame/682f3a39a0a0a05e2290606a3df26eac575df669/Makefile

Reply:
_seven_ years ago it was `hack/build-go.sh $(WHAT)`. around 5 years ago things switched more to make and it went to `hack/make-rules/build.sh` and then the dependency on `generated_files` was introduced to automatically generate those.